File: exams/codena/u3/Strategy/src/main/java/ec/edu/espe/strategy/controller/SortingContext.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.strategy.controller;

/**
 *
 * @author Daniel Codena, @CodeBreakers, ESPE
 */
public class SortingContext {

    private SortingStrategy sortingStrategy;

    public SortingStrategy sort(int size) {
        if (size < 2) {
            System.out.println("Enter at least 2 numbers");
        } else {
            if (size < 7) {
                this.setSortStrategy(new BubbleSort());
            } else {
                if (size < 11) {
                    this.setSortStrategy(new InsertionSort());
                } else {
                    this.setSortStrategy(new QuickSort());
                }
            }
        }
        return sortingStrategy;
    }

    private void setSortStrategy(SortingStrategy strategy) {
        this.sortingStrategy = strategy;
    }

    public SortingStrategy getSortingStrategy() {
        return sortingStrategy;
    }
    
    
}

File: exams/chavez/u3/Strategy/strategy/SortingContext.js

const BubbleSort = require("../strategy/BubbleSort");
const InsertionSort = require("../strategy/InsertionSort");
const QuickSort = require("../strategy/QuickSort");
class SortingContext {

    constructor() {
        this.strategy = null;
    }

    setStrategy(strategy) {
        this.strategy = strategy;
    }

    sort(array) {

        if (array.length >= 2 && array.length <= 6) {
            this.setStrategy(new BubbleSort());

        } else if (array.length >= 7 && array.length <= 10) {
            this.setStrategy(new InsertionSort());

        } else  if (array.length >=11) {
            this.setStrategy(new QuickSort());
        }

        return {
            algorithm: this.strategy.constructor.name,
            sortedArray: this.strategy.sort(array)
        };
    }

}

module.exports = SortingContext;